lib/Sema/TypeCheckCodeCompletion.cpp

```diff
@@ -1104,6 +1104,10 @@ fallbackTypeCheck(DeclContext *DC) {
 }
 
 static Type getTypeForCompletion(const constraints::Solution &S, Expr *E) {
+  if (!S.hasType(E)) {
```

Comment:
If the expression was passed to `typeCheckForCodeCompletion` it should have a type set always, I wonder if this is bug in code completion or constraint solver?

Reply:
@xedin Unfortunately it seems we're type checking the wrong expression somewhere which causes us to fail to get the type of the original code completion expr, this check at least stops us from crashing in that case.

In addition for examples like completion in a result builder, which I believe Alex is currently working on, it's possible the completion expr is in an invalid statement that gets marked an unhandled and therefore never reaches CSGen. It would be nice if we could better recover from that with holes, but for now this lets us recover.

Reply:
It sounds like the actual problem is that there is no feedback to distinguish invalid cases. It makes me think that the actual fix here might be to register invalid expressions/statements as having `ErrorType` in the constraint system.

And sorry, for some reason github didn't notify me that there was a rely here...